src/screenComponents/miniGame.h

```diff
@@ -0,0 +1,40 @@
+/** An abstract class to present an interface on which to implement (hacking)
+ *  minigames. Concrete implementations will need to implement at least their
+ *  own constructor and reset() functions, and probably the disable() function.
+ *  Calling gameComplete() registers a successful minigame with the owner
+ */
+
+#ifndef MINIGAME_H
+#define MINIGAME_H
+
+#include <vector>
+#include "hackingDialog.h"
+
+class GuiButton;
+class GuiToggleButton;
+class GuiProgressbar;
+class GuiLabel;
+class GuiPanel;
+
+class MiniGame
+{
+  public:
+    MiniGame(GuiPanel* owner, GuiHackingDialog* parent, int difficulty);
+    virtual ~MiniGame();
+
+    virtual float getProgress();
```

Comment:
If any of these are "must be implemented by subclass", it would be better to make them true abstract virtual functions:
https://en.wikibooks.org/wiki/C%2B%2B_Programming/Classes/Abstract_Classes

Reply:
I'm a bit conflicted on this, and open to suggestions. I had these as pure virtual functions originally. I decided it was nicer to have the hacking dialog have a dummy Minigamepanel when first opened, and thus needed to actually instantiate the class MiniGame, so these had to be implemented.

The more I think about it, the more I'm just going to go with an Empty panel instead of the dummy minigame, and make these abstract/pure virtual again.

Reply:
And, after looking at it, that doesn't work either, as the Reset Button would otherwise call reset() on a null pointer. There needs to be something in the game variable, and I think presenting an empty board is better to guide the user towards making a choice on which system to hack than a filled, but deactivated board. So, I think I will leave this as is.

Reply:
Ok. Good. I was just wondering :-)